### PR TITLE
load default for login.keychain-db

### PR DESCRIPTION
**Pull request by csc-EricWu**
load default for login.keychain-db

### CODE REVIEW

File: XReSign/ViewController.swift [hunk 1]
```diff
@@ -27,6 +29,9 @@ class ViewController: NSViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        if let defaultProvisioning = UserDefaults.standard.string(forKey: kLastProvisioningPathKey) {
+            textFieldProvisioningPath.stringValue = defaultProvisioning
+        }
```
Comment (xndrs) [inline XReSign/ViewController.swift]: what if this file is not exists? in this case you knowingly fill the field with a non-existing file.

File: XReSign/ViewController.swift [hunk 2]
```diff
@@ -299,7 +311,12 @@ class ViewController: NSViewController {
                               style: .critical)
                 return
             }
-            
+            if !FileManager.default.fileExists(atPath: provisioningPath) {
+                showAlertWith(title: nil,
+                message: "Provisioning not exists!",
+                style: .critical)
+                return
+            }
```
Comment (xndrs) [inline XReSign/ViewController.swift]: technically, this field is optional. if you do not use an external file, then all the data is taken from the file inside the ipa. you oblige the user to select the file all the time.

File: XReSign/ViewController.swift (hunk 1 above)
Comment (csc-EricWu) [reply XReSign/ViewController.swift]: Remember the last path to UserDefault by default，last If the file is deleted, you will get an error

File: XReSign/ViewController.swift (hunk 2 above)
Comment (csc-EricWu) [reply XReSign/ViewController.swift]: when provisioningPath is not empty ,need  provisioningPath exists sure

File: XReSign/ViewController.swift [hunk 3]
```diff
@@ -69,6 +74,13 @@ class ViewController: NSViewController {
         DispatchQueue.main.sync {
             self.keychains = map
             self.comboBoxKeychains.reloadData()
+            //load default for login.keychain-db
+            map.enumerated().forEach { arg0 in
+                let (offset, (key, _)) = arg0
+                if key.localizedStandardContains("login.") {
+                    self.comboBoxKeychains.selectItem(at: offset)
+                }
+            }
```
Comment (xndrs) [inline XReSign/ViewController.swift]: Good idea, but may be use here a script like: _"security default-keychain"_ ? User can be logged in with different keychains. this request returns the default keychain.

File: XReSign/ViewController.swift (hunk 1 above)
Comment (xndrs) [reply XReSign/ViewController.swift]: you may insert probably the wrong field into the textFieldProvisioningPath if file does not exist. this is the point. may be before check if file exists and after this assign the value to textField? if file not exists assign the empty value.

File: XReSign/ViewController.swift (hunk 2 above)
Comment (xndrs) [reply XReSign/ViewController.swift]: so why you interrupt the flow by"return"? provisioningPath can be empty. it's optional!

Comment (csc-EricWu) [reply XReSign/ViewController.swift]: You should look at the complete code. This condition determines whether the file exists when provisioningpath is not empty，Ensure the path of the last record. The file must exist

File: XReSign/ViewController.swift (hunk 3 above)
Comment (csc-EricWu) [reply XReSign/ViewController.swift]: The idea comes from iresign

**Comment by xndrs** [comment]
Thanks. Good work

**Pull request merged**